app/src/main/java/fr/free/nrw/commons/LocationPicker/LocationPickerActivity.java

```diff
@@ -0,0 +1,267 @@
+import com.mapbox.mapboxsdk.maps.MapboxMap.OnCameraMoveStartedListener;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import fr.free.nrw.commons.R;
+import org.jetbrains.annotations.NotNull;
+import timber.log.Timber;
+
+/**
+ * Helps to pick location and return the result with an intent
+ */
+public class LocationPickerActivity extends AppCompatActivity implements OnMapReadyCallback,
+    OnCameraMoveStartedListener, OnCameraIdleListener, Observer<CameraPosition> {
+
+  /**
+   * cameraPosition : position of picker
+   */
+  private CameraPosition cameraPosition;
+  /**
+   * markerImage : picker image
+   */
+  private ImageView markerImage;
+  /**
+   * mapboxMap : map
+   */
+  private MapboxMap mapboxMap;
+  /**
+   * mapView : view of the map
+   */
+  private MapView mapView;
+
+  @Override
+  protected void onCreate(@Nullable final Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    getWindow().requestFeature(Window.FEATURE_ACTION_BAR);
+    final ActionBar actionBar = getSupportActionBar();
+    if (actionBar != null) {
+      actionBar.hide();
+    }
+    setContentView(R.layout.activity_location_picker);
+
+    if (savedInstanceState == null) {
+      cameraPosition = getIntent().getParcelableExtra(LocationPickerConstants.MAP_CAMERA_POSITION);
+    }
+
+    final LocationPickerViewModel viewModel = new ViewModelProvider(this)
+        .get(LocationPickerViewModel.class);
+    viewModel.getResult().observe(this, this);
+
+    bindViews();
+    addBackButtonListener();
+    addPlaceSelectedButton();
+    getToolbarUI();
+
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  /**
+   * Clicking back button destroy locationPickerActivity
+   */
+  private void addBackButtonListener() {
+    final ImageView backButton = findViewById(R.id.mapbox_place_picker_toolbar_back_button);
+    backButton.setOnClickListener(view -> finish());
+  }
+
+  /**
+   * Binds mapView and location picker icon
+   */
+  private void bindViews() {
+    mapView = findViewById(R.id.map_view);
+    markerImage = findViewById(R.id.location_picker_image_view_marker);
+  }
+
+  /**
+   * Binds the listeners
+   */
+  private void bindListeners() {
+    mapboxMap.addOnCameraMoveStartedListener(
+        this);
+    mapboxMap.addOnCameraIdleListener(
+        this);
+  }
+
+  /**
+   * Gets toolbar color
+   */
+  private void getToolbarUI() {
+    final ConstraintLayout toolbar = findViewById(R.id.location_picker_toolbar);
+      toolbar.setBackgroundColor(getResources().getColor(R.color.primaryColor));
+  }
+
+  /**
+   * Takes action when map is ready to show
+   * @param mapboxMap map
+   */
+  @Override
+  public void onMapReady(final MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+      adjustCameraBasedOnOptions();
+      bindListeners();
+      enableLocationComponent(style);
+    });
```

Comment:
To set the style of the map, please reuse the code from `NearbyParentFragment.java`'s `onViewCreated`.
In particular, please use `R.string.map_attribution` after setting `setLogoEnabled(false)` and `setAttributionEnabled(false)` as it is done in NearbyParentFragment.
Ideally this styling code should be factorized so that it can be used by both nearby and your feature.
Thanks!

Reply:
@nicolas-raoul Are you talking about restyling only the credit part? or restyling the whole UI as it is done in NearbyParentFragment?

Reply:
Sorry for the lack of precision, the credit part is the most important. The scale and compass are good additions too. On the other hand, the part that shows the current GPS position is not relevant.